client/src/components/NewModulePopup.jsx

```diff
@@ -0,0 +1,239 @@
+import { React, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import DialogTitle from '@mui/material/DialogTitle';
+import FormLabel from '@mui/material/FormLabel';
+import {
+  ref, uploadBytesResumable,
+} from 'firebase/storage';
+import { storage } from '../pages/firebase';
+import * as api from '../api';
+import { serviceAreas } from '../constants';
+
+export default function NewModulePopup(props) {
+  const {
+    updateModule, open, handleClose, parentID,
+  } = props;
+
+  const roles = [];
+  // const [percent, setPercent] = useState(0);
+  const [fileLinks, setFileLinks] = useState([]);
+  const [title, setTitle] = useState('');
+  const [body, setBody] = useState('');
+  const [mentor, setMentor] = useState(false);
+  const [caregiver, setCaregiver] = useState(false);
+  const [serviceAreaToSelected, setServiceAreaToSelected] = useState({});
+
+  // add permissions to view module. order doesn't matter
+  if (mentor) {
+    roles.push('mentor');
+  }
+  if (caregiver) {
+    roles.push('caregiver');
+  }
+
+  useEffect(() => {
+    serviceAreas.forEach((SA) => {
+      const tempServiceAreaToSelected = { ...serviceAreaToSelected };
+      tempServiceAreaToSelected[SA] = false;
+      setServiceAreaToSelected(tempServiceAreaToSelected);
+    });
+  }, [serviceAreas]);
+
+  // TODO: Move to backend, figure out how to maintain setPercent once it is moved to the backedn and sent back as a promise chain
+  // upload file to Firebase:
+  const handleUpload = (file) => {
+    const fileName = file.name;
+    const storageRef = ref(storage, `/files/${fileName}`);
+    const uploadTask = uploadBytesResumable(storageRef, file);
+
+    uploadTask.on(
+      'state_changed',
+      (snapshot) => {
+        const p = Math.round(
+          (snapshot.bytesTransferred / snapshot.totalBytes) * 100,
+        );
+
+        // update progress
+        // setPercent(p);
```

Comment:
file upload seems to be working perfectly so you can remove the setPercent line and the console.log!